src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 export interface Figure {
-
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  color: Color;
+
+  sides: number[];
+
+  constructor(color: Color, a: number, b: number, c: number) {
```

Comment:
you can access Figure types with this syntax: `Figure['typeName']` - you won't have to retype everything again